frontend/src/Components/ColorModeSwitch.logic.ts

```diff
@@ -0,0 +1,28 @@
+import { useLayoutEffect, useState } from 'preact/hooks';
```

Comment:
Could you move the component files into a folder `ColorModeSwitch`, similar to `LoginPage` and `ResultsPage`? The ultimate goal is to move all components into folders, unfortunately the review has not been done for #76 yet.

Reply:
Yes, I will do. I will also take a look at your PR, which looks good at first glance, if you are also content with my review.